processor/src/lib.rs

```diff
@@ -147,6 +147,30 @@ where
     VmStateIterator::new(process, result)
 }
 
+/// Returns a tuple with VM state iterator and trace lengths summary.
+pub fn execute_iter_with_trace_len<H>(
+    program: &Program,
+    stack_inputs: StackInputs,
+    host: H,
+) -> (VmStateIterator, TraceLenSummary)
```

Comment:
I would prefer not to introduce a new method but just to add `trace_len_summary` to the `VmStateIterator` struct.